crates/sp-domains-fraud-proof/src/host_functions.rs

```diff
@@ -257,6 +257,25 @@ where
             .ok()
     }
 
+    fn is_decodable_extrinsic(
+        &self,
+        consensus_block_hash: H256,
+        domain_id: DomainId,
+        opaque_extrinsic: OpaqueExtrinsic,
+    ) -> Option<bool> {
+        let runtime_code = self.get_domain_runtime_code(consensus_block_hash, domain_id)?;
+        let domain_stateless_runtime =
+            StatelessRuntime::<DomainBlock, _>::new(self.executor.clone(), runtime_code.into());
+
+        let encoded_extrinsic = opaque_extrinsic.encode();
+        let extrinsic =
+            <DomainBlock as BlockT>::Extrinsic::decode(&mut encoded_extrinsic.as_slice()).ok()?;
```

Comment:
Why decode into Opaque and then Encoding only to be Decoded by the actual type?
Maybe just sent in the `Vec` as it and try decoding the final type instead. Intermediate encoding and decoding is redundant

Reply:
This is how the `UndecodableTx` is defined today: https://github.com/subspace/subspace/blob/main/domains/client/block-preprocessor/src/lib.rs#L267. It basically indicates that the concrete type could not be decoded from the opaque extrinsic. 

But anyways, looking at the `OpaqueExtrinsic` code, there doesn't seem to be a way to extract the inner `Vec`/`self.0` to avoid the encode/decode